[DebugMode] add DebugInterpreter for fx.Node metadata in  aot_eager compiled regions

pianpwk opened the pull request:
One difficulty with DebugMode has been contextualizing logs. For regions compiled with `backend="eager/aot_eager"`, this PR switches the runtime to use `DebugInterpreter(gm).run(...)` instead of the compiled code, so `run_node()` has access to node metadata, and can contextualize logs.

The is enabled with `DebugMode(run_compile_with_interpreter=True)`, which defaults to False in case the interpreter introduces divergence, but ideally it'd be on.

This PR uses nn_module_stack to add `[nn.Module]` logs for compiled regions. Example for forward + backward with aot_eager:
```
  [aot_eager region (compile)] enter
    [nn.Mod (compile)] L['self'].foo
      [nn.Mod (compile)] L['self'].foo.l1
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::addmm(t: f32[4], t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
      aten::relu(t: f32[4, 4])  ->  t: f32[4, 4]
      [nn.Mod (compile)] L['self'].foo.l2
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::addmm(t: f32[4], t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
    [nn.Mod (compile)] L['self'].bar
      aten::add.Tensor(t: f32[4, 4], 2.0)  ->  t: f32[4, 4]
      [nn.Mod (compile)] L['self'].bar.l3
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::addmm(t: f32[4], t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
  [aot_eager region (compile)] exit
    aten::sum(t: f32[4, 4])  ->  t: f32[]
    aten::ones_like(t: f32[], pin_memory=False, memory_format=torch.preserve_format)  ->  t: f32[]
    aten::expand(t: f32[], [4, 4])  ->  t: f32[4, 4]
    aten::clone(t: f32[4, 4], memory_format=torch.contiguous_format)  ->  t: f32[4, 4]
  [aot_eager region (compile)] enter
    [nn.Mod (compile)] L['self'].bar
      [nn.Mod (compile)] L['self'].bar.l3
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::mm(t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::mm(t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::sum.dim_IntList(t: f32[4, 4], [0], True)  ->  t: f32[1, 4]
        aten::view(t: f32[1, 4], [4])  ->  t: f32[4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
    [nn.Mod (compile)] L['self'].foo
      [nn.Mod (compile)] L['self'].foo.l2
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::mm(t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::mm(t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::sum.dim_IntList(t: f32[4, 4], [0], True)  ->  t: f32[1, 4]
        aten::view(t: f32[1, 4], [4])  ->  t: f32[4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
      aten::detach(t: f32[4, 4])  ->  t: f32[4, 4]
      aten::detach(t: f32[4, 4])  ->  t: f32[4, 4]
      aten::threshold_backward(t: f32[4, 4], t: f32[4, 4], 0)  ->  t: f32[4, 4]
      [nn.Mod (compile bwd)] L['self'].foo.l1
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::mm(t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::sum.dim_IntList(t: f32[4, 4], [0], True)  ->  t: f32[1, 4]
        aten::view(t: f32[1, 4], [4])  ->  t: f32[4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
  [aot_eager region (compile)] exit
```

The nn.Module tracking should also work nicely with regional compilation; DebugMode now maintains its own nn_module_stack tracking.

Stack from [ghstack](https://github.com/ezyang/ghstack/tree/0.13.0) (oldest at bottom):
* #170126
* __->__ #170125



cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @wenzhe-nrv @jiayisunx @kadeng @chauhang @amjames @Lucaskabela @jataylo

pianpwk committed:
DebugInterpreter for eager/aot_eager, w/ nn.Module calls

[ghstack-poisoned]

yushangdi commented on torch/utils/_debug_mode.py:
```diff
@@ -668,6 +637,63 @@ def _annotate_fake(tag: str) -> None:
+        result = super().run(*args)
+
+        # reset nn.Module stack to pre-compiled region value
+        assert len(self.mode.current_nn_module_stack) >= len(self.base_nn_module_stack)
```
might be good to emit a warning rather than assert? if there is a bug, users can still see a dump with the rest of the info.

pianpwk committed:
Update on "[DebugMode] add DebugInterpreter for fx.Node metadata in  eager/aot_eager compiled regions"

One difficulty with DebugMode has been contextualizing logs. For regions compiled with `backend="eager/aot_eager"`, this PR switches the runtime to use `DebugInterpreter(gm).run(...)` instead of the compiled code, so `run_node()` has access to node metadata, and can contextualize logs.

The is enabled with `DebugMode(run_compile_with_interpreter=True)`, which defaults to False in case the interpreter introduces divergence, but ideally it'd be on.

This PR uses nn_module_stack to add `[nn.Module]` logs for compiled regions. Example for forward + backward with aot_eager:
```
  [aot_eager region (compile)] enter
    [nn.Mod (compile)] L['self'].foo
      [nn.Mod (compile)] L['self'].foo.l1
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::addmm(t: f32[4], t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
      aten::relu(t: f32[4, 4])  ->  t: f32[4, 4]
      [nn.Mod (compile)] L['self'].foo.l2
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::addmm(t: f32[4], t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
    [nn.Mod (compile)] L['self'].bar
      aten::add.Tensor(t: f32[4, 4], 2.0)  ->  t: f32[4, 4]
      [nn.Mod (compile)] L['self'].bar.l3
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::addmm(t: f32[4], t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
  [aot_eager region (compile)] exit
    aten::sum(t: f32[4, 4])  ->  t: f32[]
    aten::ones_like(t: f32[], pin_memory=False, memory_format=torch.preserve_format)  ->  t: f32[]
    aten::expand(t: f32[], [4, 4])  ->  t: f32[4, 4]
    aten::clone(t: f32[4, 4], memory_format=torch.contiguous_format)  ->  t: f32[4, 4]
  [aot_eager region (compile)] enter
    [nn.Mod (compile)] L['self'].bar
      [nn.Mod (compile)] L['self'].bar.l3
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::mm(t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::mm(t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::sum.dim_IntList(t: f32[4, 4], [0], True)  ->  t: f32[1, 4]
        aten::view(t: f32[1, 4], [4])  ->  t: f32[4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
    [nn.Mod (compile)] L['self'].foo
      [nn.Mod (compile)] L['self'].foo.l2
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::mm(t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::mm(t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::sum.dim_IntList(t: f32[4, 4], [0], True)  ->  t: f32[1, 4]
        aten::view(t: f32[1, 4], [4])  ->  t: f32[4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
      aten::detach(t: f32[4, 4])  ->  t: f32[4, 4]
      aten::detach(t: f32[4, 4])  ->  t: f32[4, 4]
      aten::threshold_backward(t: f32[4, 4], t: f32[4, 4], 0)  ->  t: f32[4, 4]
      [nn.Mod (compile bwd)] L['self'].foo.l1
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::mm(t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::sum.dim_IntList(t: f32[4, 4], [0], True)  ->  t: f32[1, 4]
        aten::view(t: f32[1, 4], [4])  ->  t: f32[4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
  [aot_eager region (compile)] exit
```

The nn.Module tracking should also work nicely with regional compilation; DebugMode now maintains its own nn_module_stack tracking.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx kadeng chauhang amjames Lucaskabela jataylo

[ghstack-poisoned]

pianpwk committed:
Update on "[DebugMode] add DebugInterpreter for fx.Node metadata in  eager/aot_eager compiled regions"

One difficulty with DebugMode has been contextualizing logs. For regions compiled with `backend="eager/aot_eager"`, this PR switches the runtime to use `DebugInterpreter(gm).run(...)` instead of the compiled code, so `run_node()` has access to node metadata, and can contextualize logs.

The is enabled with `DebugMode(run_compile_with_interpreter=True)`, which defaults to False in case the interpreter introduces divergence, but ideally it'd be on.

This PR uses nn_module_stack to add `[nn.Module]` logs for compiled regions. Example for forward + backward with aot_eager:
```
  [aot_eager region (compile)] enter
    [nn.Mod (compile)] L['self'].foo
      [nn.Mod (compile)] L['self'].foo.l1
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::addmm(t: f32[4], t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
      aten::relu(t: f32[4, 4])  ->  t: f32[4, 4]
      [nn.Mod (compile)] L['self'].foo.l2
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::addmm(t: f32[4], t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
    [nn.Mod (compile)] L['self'].bar
      aten::add.Tensor(t: f32[4, 4], 2.0)  ->  t: f32[4, 4]
      [nn.Mod (compile)] L['self'].bar.l3
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::addmm(t: f32[4], t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
  [aot_eager region (compile)] exit
    aten::sum(t: f32[4, 4])  ->  t: f32[]
    aten::ones_like(t: f32[], pin_memory=False, memory_format=torch.preserve_format)  ->  t: f32[]
    aten::expand(t: f32[], [4, 4])  ->  t: f32[4, 4]
    aten::clone(t: f32[4, 4], memory_format=torch.contiguous_format)  ->  t: f32[4, 4]
  [aot_eager region (compile)] enter
    [nn.Mod (compile)] L['self'].bar
      [nn.Mod (compile)] L['self'].bar.l3
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::mm(t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::mm(t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::sum.dim_IntList(t: f32[4, 4], [0], True)  ->  t: f32[1, 4]
        aten::view(t: f32[1, 4], [4])  ->  t: f32[4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
    [nn.Mod (compile)] L['self'].foo
      [nn.Mod (compile)] L['self'].foo.l2
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::mm(t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::mm(t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::sum.dim_IntList(t: f32[4, 4], [0], True)  ->  t: f32[1, 4]
        aten::view(t: f32[1, 4], [4])  ->  t: f32[4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
      aten::detach(t: f32[4, 4])  ->  t: f32[4, 4]
      aten::detach(t: f32[4, 4])  ->  t: f32[4, 4]
      aten::threshold_backward(t: f32[4, 4], t: f32[4, 4], 0)  ->  t: f32[4, 4]
      [nn.Mod (compile bwd)] L['self'].foo.l1
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::mm(t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::sum.dim_IntList(t: f32[4, 4], [0], True)  ->  t: f32[1, 4]
        aten::view(t: f32[1, 4], [4])  ->  t: f32[4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
  [aot_eager region (compile)] exit
```

The nn.Module tracking should also work nicely with regional compilation; DebugMode now maintains its own nn_module_stack tracking.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx kadeng chauhang amjames Lucaskabela jataylo

[ghstack-poisoned]

pianpwk committed:
Update

[ghstack-poisoned]

pianpwk committed:
Update

[ghstack-poisoned]

pianpwk commented:
@pytorchbot merge

pianpwk commented:
@pytorchbot merge -i

pianpwk committed:
Update

[ghstack-poisoned]

pianpwk committed:
Update

[ghstack-poisoned]

pianpwk committed:
Update

[ghstack-poisoned]

pianpwk committed:
Update

[ghstack-poisoned]

pianpwk committed:
Update on "[DebugMode] add DebugInterpreter for fx.Node metadata in  eager/aot_eager compiled regions"

One difficulty with DebugMode has been contextualizing logs. For regions compiled with `backend="eager/aot_eager"`, this PR switches the runtime to use `DebugInterpreter(gm).run(...)` instead of the compiled code, so `run_node()` has access to node metadata, and can contextualize logs.

The is enabled with `DebugMode(run_compile_with_interpreter=True)`, which defaults to False in case the interpreter introduces divergence, but ideally it'd be on.

This PR uses nn_module_stack to add `[nn.Module]` logs for compiled regions. Example for forward + backward with aot_eager:
```
  [aot_eager region (compile)] enter
    [nn.Mod (compile)] L['self'].foo
      [nn.Mod (compile)] L['self'].foo.l1
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::addmm(t: f32[4], t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
      aten::relu(t: f32[4, 4])  ->  t: f32[4, 4]
      [nn.Mod (compile)] L['self'].foo.l2
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::addmm(t: f32[4], t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
    [nn.Mod (compile)] L['self'].bar
      aten::add.Tensor(t: f32[4, 4], 2.0)  ->  t: f32[4, 4]
      [nn.Mod (compile)] L['self'].bar.l3
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::addmm(t: f32[4], t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
  [aot_eager region (compile)] exit
    aten::sum(t: f32[4, 4])  ->  t: f32[]
    aten::ones_like(t: f32[], pin_memory=False, memory_format=torch.preserve_format)  ->  t: f32[]
    aten::expand(t: f32[], [4, 4])  ->  t: f32[4, 4]
    aten::clone(t: f32[4, 4], memory_format=torch.contiguous_format)  ->  t: f32[4, 4]
  [aot_eager region (compile)] enter
    [nn.Mod (compile)] L['self'].bar
      [nn.Mod (compile)] L['self'].bar.l3
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::mm(t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::mm(t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::sum.dim_IntList(t: f32[4, 4], [0], True)  ->  t: f32[1, 4]
        aten::view(t: f32[1, 4], [4])  ->  t: f32[4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
    [nn.Mod (compile)] L['self'].foo
      [nn.Mod (compile)] L['self'].foo.l2
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::mm(t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::mm(t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::sum.dim_IntList(t: f32[4, 4], [0], True)  ->  t: f32[1, 4]
        aten::view(t: f32[1, 4], [4])  ->  t: f32[4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
      aten::detach(t: f32[4, 4])  ->  t: f32[4, 4]
      aten::detach(t: f32[4, 4])  ->  t: f32[4, 4]
      aten::threshold_backward(t: f32[4, 4], t: f32[4, 4], 0)  ->  t: f32[4, 4]
      [nn.Mod (compile bwd)] L['self'].foo.l1
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::mm(t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::sum.dim_IntList(t: f32[4, 4], [0], True)  ->  t: f32[1, 4]
        aten::view(t: f32[1, 4], [4])  ->  t: f32[4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
  [aot_eager region (compile)] exit
```

The nn.Module tracking should also work nicely with regional compilation; DebugMode now maintains its own nn_module_stack tracking.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx kadeng chauhang amjames Lucaskabela jataylo

[ghstack-poisoned]

pianpwk committed:
Update on "[DebugMode] add DebugInterpreter for fx.Node metadata in  aot_eager compiled regions"

One difficulty with DebugMode has been contextualizing logs. For regions compiled with `backend="eager/aot_eager"`, this PR switches the runtime to use `DebugInterpreter(gm).run(...)` instead of the compiled code, so `run_node()` has access to node metadata, and can contextualize logs.

The is enabled with `DebugMode(run_compile_with_interpreter=True)`, which defaults to False in case the interpreter introduces divergence, but ideally it'd be on.

This PR uses nn_module_stack to add `[nn.Module]` logs for compiled regions. Example for forward + backward with aot_eager:
```
  [aot_eager region (compile)] enter
    [nn.Mod (compile)] L['self'].foo
      [nn.Mod (compile)] L['self'].foo.l1
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::addmm(t: f32[4], t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
      aten::relu(t: f32[4, 4])  ->  t: f32[4, 4]
      [nn.Mod (compile)] L['self'].foo.l2
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::addmm(t: f32[4], t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
    [nn.Mod (compile)] L['self'].bar
      aten::add.Tensor(t: f32[4, 4], 2.0)  ->  t: f32[4, 4]
      [nn.Mod (compile)] L['self'].bar.l3
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::addmm(t: f32[4], t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
  [aot_eager region (compile)] exit
    aten::sum(t: f32[4, 4])  ->  t: f32[]
    aten::ones_like(t: f32[], pin_memory=False, memory_format=torch.preserve_format)  ->  t: f32[]
    aten::expand(t: f32[], [4, 4])  ->  t: f32[4, 4]
    aten::clone(t: f32[4, 4], memory_format=torch.contiguous_format)  ->  t: f32[4, 4]
  [aot_eager region (compile)] enter
    [nn.Mod (compile)] L['self'].bar
      [nn.Mod (compile)] L['self'].bar.l3
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::mm(t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::mm(t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::sum.dim_IntList(t: f32[4, 4], [0], True)  ->  t: f32[1, 4]
        aten::view(t: f32[1, 4], [4])  ->  t: f32[4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
    [nn.Mod (compile)] L['self'].foo
      [nn.Mod (compile)] L['self'].foo.l2
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::mm(t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::mm(t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::sum.dim_IntList(t: f32[4, 4], [0], True)  ->  t: f32[1, 4]
        aten::view(t: f32[1, 4], [4])  ->  t: f32[4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
      aten::detach(t: f32[4, 4])  ->  t: f32[4, 4]
      aten::detach(t: f32[4, 4])  ->  t: f32[4, 4]
      aten::threshold_backward(t: f32[4, 4], t: f32[4, 4], 0)  ->  t: f32[4, 4]
      [nn.Mod (compile bwd)] L['self'].foo.l1
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::mm(t: f32[4, 4], t: f32[4, 4])  ->  t: f32[4, 4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
        aten::sum.dim_IntList(t: f32[4, 4], [0], True)  ->  t: f32[1, 4]
        aten::view(t: f32[1, 4], [4])  ->  t: f32[4]
        aten::t(t: f32[4, 4])  ->  t: f32[4, 4]
  [aot_eager region (compile)] exit
```

The nn.Module tracking should also work nicely with regional compilation; DebugMode now maintains its own nn_module_stack tracking.




cc voznesenskym penguinwu EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng wenzhe-nrv jiayisunx kadeng chauhang amjames Lucaskabela jataylo

[ghstack-poisoned]

pianpwk commented:
@pytorchbot merge

pianpwk committed:
Update

[ghstack-poisoned]

pianpwk committed:
Update

[ghstack-poisoned]
